File: food_delivery_flutter/lib/Model/Product.dart
import 'dart:ffi';

class Product{
  final int product_id;
  final String name;
  final String description;
  final double price;
  final int category_id;
  final String imageURL;

  Product({required this.product_id, this.name='', this.description='', this.price=0, required this.category_id, this.imageURL=''});

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      product_id: json['product_id'],
      name: json['name'],
      description: json['description'],
      price: json['price'],
      category_id: json['category_id'],
      imageURL: json['imageURL'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'product_id': product_id,
      'name': name,
      'description': description,
      'price': price,
      'category_id': category_id,
      'imageURL': imageURL,
    };
  }

  factory Product.toJson(Map<String, dynamic> json) {
    return Product(
      product_id: json['product_id'],
      name: json['name'],
      description: json['description'],
      price: json['price'],
      category_id: json['category_id'],
      imageURL: json['imageURL'],
    );
  }
}

File: food_delivery/src/main/java/com/datn/food_delivery/models/Product.java
package com.datn.food_delivery.models;

import jakarta.persistence.*;

//@Entity
//@Table(name = "Product")
public class Product {
//    @Id
//    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long product_id;
    private String name;
    private String description;
    private double price;
    private Long category_id;
    private String imageURL;


    public Product() {
    }

    public Product(Long product_id, String name, String description, double price, Long category_id, String imageURL) {
        this.product_id = product_id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.category_id = category_id;
        this.imageURL = imageURL;
    }

    public Product(Long product_id, String name, String description, double price, Long category_id) {
        this.product_id = product_id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.category_id = category_id;
    }

    public long getProduct_id() {
        return product_id;
    }

    public void setProduct_id(long product_id) {
        this.product_id = product_id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public Long getCategory_id() {
        return category_id;
    }

    public void setCategory_id(Long category_id) {
        this.category_id = category_id;
    }

    public String getImageURL() {
        return imageURL;
    }

    public void setImageURL(String imageURL) {
        this.imageURL = imageURL;
    }


}
